common/data_structure.go

```diff
@@ -2,4 +2,53 @@ package common
 
 type FileAttributes struct {
 	Size uint
-}
\ No newline at end of file
+}
+
+type BlockMetadata struct {
+	Size uint
+}
+
+type BlockReport struct {
+	Endpoint      string
+	BlockMetadata map[string]BlockMetadata
+}
+
+type Heartbeat struct {
+	Endpoint string
+}
+
+type ReadFileRequest struct {
+	FileName   string
+	BlockRange [2]uint
```

Comment:
Since this array has fixed number of elements and each element has a fixed meaning, I suggest extracting these elements out, and adding two fields called `begin` and `end` instead. Make sure that you added comments about whether they are inclusive or exclusive. By convention, begin is inclusive while end is exclusive.